applications/segment_everything/setup.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# Get the current working directory
+CWD=$(pwd)
+
+# create the data/segment_everything directory if it does not exist
+if [ ! -d "data/segment_everything" ]; then
+    mkdir -p data/segment_everything
+fi
+DATA_DIR="$CWD/data/segment_everything"
+
+# save the current working directory
+pushd $CWD
+# Create the directory based on the current working directory
+mkdir -p "$DATA_DIR/software/tensorrt"
+cd "$DATA_DIR/software/tensorrt"
+
+# Detect the platform
+ARCH=$(uname -m)
+
+# Set the URL and filename based on the architecture
+if [ "$ARCH" == "aarch64" ]; then
+    FILENAME="nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0_1.0-1_arm64.deb"
+else
+    FILENAME="nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0_1.0-1_amd64.deb"
+fi
+
+URL="https://developer.nvidia.com/downloads/compute/machine-learning/tensorrt/secure/8.6.1/local_repos/$FILENAME"
+
+# Check if the file has been downloaded already
+if [ ! -f "$FILENAME" ]; then
+    wget $URL
+else
+    echo "$FILENAME already exists, skipping download."
+fi
+
+popd
+
+# sam_trt_light
+# download the repository if it does not exist
+cd $DATA_DIR
+if [ ! -d "sam_trt_light" ]; then
+    git clone https://github.com/maximilianofir/sam_trt_light.git
```

Comment:
Ideally we should not download from a personal GitHub repo. Is this a fork, or are there components that could be added to this app folder? Are the SAM terms permissive such that we could host and distribute the encoder/decode engine files ourselves?